### DISCUSSION TITLE
Menu Components: Add option for templating labels only

### DISCUSSION THREAD

**Discussion by scurk1415** [primefaces/community#436] [PrimeNG]
16.6.0 Added the support for custom templates for menus and menuBars. One big disadvantage of this item template is that we have to template everything ourselves. That includes the icons and also the side caret for multilevel items (in case of MenuBar). We want to keep the primeNG default template, but just transform the label.

It would be nice to have an option for just the label template, something like this (same for normal Menu)
```html
<p-menubar [model]="items">
    <ng-template pTemplate="label" let-item>
        {{ item.label | translate }}
    </ng-template>
</p-menubar>
```

My motivation for this is the usage of pipes for the label. This is especially useful for translations as we currently have to prepare the whole menu items list when the language changes (i don't know about a better solution)

**Comment by cetincakiroglu** [comment]
Hi,

We've recently implemented item template for menu components. You can customize the label by using item templating as you wish.

**Reply by scurk1415** [reply]
Yeah but as I said, you have to template the whole item. That includes the icon, the ">" sign (MenuBar multi layer items) and everything else.

If i want to do this `{{ item.label | translate }}` for the MenuBar item I would have to copy your whole item code (which I guess is all this code? https://github.com/primefaces/primeng/blob/master/src/app/components/menubar/menubar.ts#L102).

But all i want is to transform the label, the other functionality should stay as it is
